packages/core-ethereum/src/transaction-manager.ts

```diff
@@ -72,27 +101,42 @@ class TranscationManager {
     }
 
     const hash = Array.from(this.payloads.keys())[index]
-    if (!this.mined.get(hash) && BigNumber.from(this.pending.get(hash).gasPrice).lt(BigNumber.from(gasPrice))) {
+    if (
+      !this.mined.get(hash) &&
+      BigNumber.from((this.pending.get(hash) ?? this.queuing.get(hash)).gasPrice).lt(BigNumber.from(gasPrice))
+    ) {
       return [false, hash]
     }
     return [true, hash]
   }
 
   /**
-   * Adds transaction in pending
+   * Adds queuing transaction
    * @param hash transaction hash
    * @param transaction object
    */
-  public addToPending(
+  public addToQueuing(
     hash: string,
     transaction: Omit<Transaction, 'createdAt'>,
     transactionPayload: TransactionPayload
   ): void {
-    if (this.pending.has(hash)) return
+    if (this.queuing.has(hash)) return
 
-    log('Adding pending transaction %s %i', hash, transaction.nonce)
+    log('Adding queuing transaction %s %i', hash, transaction.nonce)
     this.payloads.set(hash, transactionPayload)
-    this.pending.set(hash, { nonce: transaction.nonce, createdAt: this._getTime(), gasPrice: transaction.gasPrice })
+    this.queuing.set(hash, { nonce: transaction.nonce, createdAt: 0, gasPrice: transaction.gasPrice })
+  }
+
+  /**
+   * Moves transaction from queuing to pending
+   * @param hash transaction hash
+   */
+  public moveFromQueuingToPending(hash: string): void {
+    if (!this.queuing.has(hash)) return
+
+    log('Moving transaction to pending %s', hash)
+    this.pending.set(hash, { ...this.queuing.get(hash), createdAt: this._getTime() })
+    this.queuing.delete(hash)
```

Comment:
should this be deleted in `public remove(hash: string): void` too?

Reply:
Yes. Also added missing handling in `public moveToConfirmed(hash: string)`